### PR TITLE
Respect client idle timeout setting

### PR DESCRIPTION
**Pull request by rosstimothy**
Client connections were never using the `srv.TrackingReadConn` created by the `srv.ConnectionMonitor` which caused all connections to appear idle and be disconnected whenever the client idle timeout was reached.

`(srv.ConnectionMonitor) MonitorConn` now returns the provided connection which is wrapped in the tracking connection. Callers must ensure that the returned connection is used instead of the connection passed in to `MonitorConn`.

Fixes #27392

### CODE REVIEW

**Review by Joerger — APPROVED** [review]

File: integration/integration_test.go [hunk 1]
```diff
@@ -128,6 +128,7 @@ func TestIntegrations(t *testing.T) {
 	t.Run("BPFExec", suite.bind(testBPFExec))
 	t.Run("BPFInteractive", suite.bind(testBPFInteractive))
 	t.Run("BPFSessionDifferentiation", suite.bind(testBPFSessionDifferentiation))
+	t.Run("ClientIdleConnection", suite.bind(testClientIdleConnection))
```
Comment (jakule) [inline integration/integration_test.go:131]: Not related to the PR itself, are they any reasons why we are still adding the system tests in that one master test suite?

File: integration/integration_test.go [hunk 2]
```diff
@@ -1809,6 +1810,86 @@ type disconnectTestCase struct {
 	verifyError errorVerifier
 }
 
+type repeatingReader struct {
```
Comment (jakule) [inline integration/integration_test.go:1813]: Can you add a comment on what this struct is doing?

File: integration/integration_test.go [hunk 3]
```diff
@@ -1809,6 +1810,86 @@ type disconnectTestCase struct {
...
+	s        string
+	interval time.Duration
+}
+
+func (r repeatingReader) Read(p []byte) (int, error) {
```
Comment (jakule) [inline integration/integration_test.go:1818]: This function never returns `EOF` or any other error. How can you know if the stream is closed?

**Review by jakule — APPROVED** [review]
Few nits, LGTM

**Pull request merged**